test/with_dummyserver/test_socketlevel.py

```diff
@@ -1393,3 +1393,36 @@ def socket_handler(listener):
         self.assertEqual(len(data), 0)
 
         done_event.set()
+
+
+class TestRetryPoolSizeDrainFail(SocketDummyServerTestCase):
+
+    def test_pool_size_retry_drain_fail(self):
+        def socket_handler(listener):
+            while 1:
```

Comment:
We don't want this to run forever: can we make sure it only runs the number of times we expect it to? (Should be twice)